examples/run_generation.py

```diff
@@ -210,28 +213,51 @@ def main():
     requires_preprocessing = args.model_type in PREPROCESSING_FUNCTIONS.keys()
     if requires_preprocessing:
         prepare_input = PREPROCESSING_FUNCTIONS.get(args.model_type)
-        prompt_text = prepare_input(args, model, tokenizer, prompt_text)
-    encoded_prompt = tokenizer.encode(prompt_text, add_special_tokens=False, return_tensors="pt")
+        preprocessed_prompt_text = prepare_input(args, model, tokenizer, prompt_text)
+        encoded_prompt = tokenizer.encode(preprocessed_prompt_text, add_special_tokens=False, return_tensors="pt")
+    else:
+        encoded_prompt = tokenizer.encode(prompt_text, add_special_tokens=False, return_tensors="pt")
     encoded_prompt = encoded_prompt.to(args.device)
 
     output_sequences = model.generate(
         input_ids=encoded_prompt,
-        max_length=args.length,
+        max_length=args.length + len(encoded_prompt[0]),
         temperature=args.temperature,
         top_k=args.k,
         top_p=args.p,
         repetition_penalty=args.repetition_penalty,
         do_sample=True,
+        num_return_sequences=args.num_return_sequences,
     )
 
-    # Batch size == 1. to add more examples please use num_return_sequences > 1
-    generated_sequence = output_sequences[0].tolist()
-    text = tokenizer.decode(generated_sequence, clean_up_tokenization_spaces=True)
-    text = text[: text.find(args.stop_token) if args.stop_token else None]
+    # Remove the batch dimension when returning multiple sequences
+    if len(output_sequences.shape) > 2:
+        output_sequences.squeeze_()
+
+    generated_sequences = []
+
+    for generated_sequence in output_sequences:
+        print("=== GENERATED SEQUENCE ===")
+        generated_sequence = generated_sequence.tolist()
+
+        # Remove "-1" tokens which are used for padding
+        generated_sequence = [token for token in generated_sequence if token != -1]
+
+        # Decode text
+        text = tokenizer.decode(generated_sequence, clean_up_tokenization_spaces=True)
+
+        # Remove all text after the stop token
+        text = text[: text.find(args.stop_token) if args.stop_token else None]
+
+        # Add the prompt at the beginning of the sequence. Remove the excess text that was used for pre-processing
+        total_sequence = (
+            prompt_text + text[len(tokenizer.decode(encoded_prompt[0], clean_up_tokenization_spaces=True)) :]
+        )
 
-    print(text)
+        generated_sequences.append(total_sequence)
+        print(total_sequence)
 
-    return text
+    return generated_sequences if len(generated_sequences) > 1 else generated_sequences[0]
```

Comment:
The proposed changes have been added to PR #2885 so that everything is included in this PR.

Reply:
Great!

Reply:
If I understand it correctly, this will either return a string (if only one item is in there), or an iterable when there are more? I don't think this is good practice. For usability's sake, I would expect the output to always be the same type. Abstraction can be made by the user.

(But of course, if members agree on this then that's how it is. :))

Reply:
Agree! Changed it to always return an iterable (see PR #2885)